public/app/api/clients/folder/v1beta1/hooks.ts

```diff
@@ -62,6 +70,91 @@ function getFolderUrl(uid: string, title: string): string {
   return `${config.appSubUrl}/dashboards/f/${uid}/${slug}`;
 }
 
+const combineFolderResponses = (
+  folder: Folder,
+  legacyFolder: FolderDTO,
+  parents: FolderInfo[],
+  userDisplay?: DisplayList
+) => {
+  const updatedBy = folder.metadata.annotations?.[AnnoKeyUpdatedBy];
+  const createdBy = folder.metadata.annotations?.[AnnoKeyCreatedBy];
+
+  const newData: FolderDTO = {
+    canAdmin: legacyFolder.canAdmin,
+    canDelete: legacyFolder.canDelete,
+    canEdit: legacyFolder.canEdit,
+    canSave: legacyFolder.canSave,
+    accessControl: legacyFolder.accessControl,
+    createdBy: (createdBy && userDisplay?.display[userDisplay?.keys.indexOf(createdBy)]?.displayName) || 'Anonymous',
+    updatedBy: (updatedBy && userDisplay?.display[userDisplay?.keys.indexOf(updatedBy)]?.displayName) || 'Anonymous',
+    ...appPlatformFolderToLegacyFolder(folder),
+  };
+
+  if (parents.length) {
+    newData.parents = parents
+      .filter((i) => i.name !== folder.metadata.name)
+      .map(({ name, title }) => ({
+        title: title,
+        uid: name,
+        // No idea how to make slug, on the server it uses a go lib: https://github.com/grafana/grafana/blob/aac66e91198004bc044754105e18bfff8fbfd383/pkg/infra/slugify/slugify.go#L56
+        // Don't think slug is needed for the URL to work though
+        url: getFolderUrl(name, title),
+      }));
+  }
+
+  return newData;
+};
+
+export async function getFolderByUidFacade(uid: string): Promise<FolderDTO> {
+  const isVirtualFolder = uid && [GENERAL_FOLDER_UID, config.sharedWithMeFolderUID].includes(uid);
+  // We need the legacy API call regardless, for now
+  const legacyApiCall = dispatch(browseDashboardsAPI.endpoints.getFolder.initiate(uid));
+
+  const shouldUseAppPlatformAPI = Boolean(config.featureToggles.foldersAppPlatformAPI);
+  if (shouldUseAppPlatformAPI) {
+    let virtualFolderResponse;
+    if (isVirtualFolder) {
+      virtualFolderResponse = GENERAL_FOLDER_UID === uid ? rootFolder : sharedWithMeFolder;
+    }
+
+    const responses = await Promise.all([
+      // We still need to call legacy endpoints for access control metadata
+      legacyApiCall,
+      isVirtualFolder
+        ? Promise.resolve({ data: virtualFolderResponse })
+        : dispatch(folderAPIv1beta1.endpoints.getFolder.initiate({ name: uid })),
+      dispatch(folderAPIv1beta1.endpoints.getFolderParents.initiate({ name: uid })),
+    ]);
+
+    const [legacyFolderResponse, folderResponse, parentsResponse] = responses;
+
+    if (!folderResponse?.data || !legacyFolderResponse?.data || !parentsResponse?.data) {
+      throw new Error('One of the folder responses is undefined');
+    }
+
+    const userKeys = getUserKeys(folderResponse.data);
+    let userResponse;
+    if (userKeys.length) {
+      userResponse = await dispatch(iamAPIv0alpha1.endpoints.getDisplayMapping.initiate({ key: userKeys }));
```

Comment:
is there smth like `skipToken` for this so we don't need to use the if statement?